### PR TITLE
More detailed documentation.

### PR DESCRIPTION
**Pull request by zhouzq-thu**
# Example

screen shot 2018-01-01 at 10 48 38 pm
screen shot 2018-01-01 at 10 48 51 pm

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -348,11 +387,19 @@ class BCELoss(_WeightedLoss):
     r"""Creates a criterion that measures the Binary Cross Entropy
     between the target and the output:
 
-    .. math:: loss(o, t) = - 1/n \sum_i (t[i] * log(o[i]) + (1 - t[i]) * log(1 - o[i]))
+    The loss can be described as:
 
-    or in the case of the weight argument being specified:
+    .. math::
+        \ell(x, y) = L = \{l_1,\dots,l_N\}^\top, \quad
+        l_n = - w_n \left[ t_n \cdot \log x_n + (1 - t_n) \cdot \log (1 - x_n) \right],
```
Comment (apaszke) [inline torch/nn/modules/loss.py]: Can you change `t_n` to `y_n` to make it more consistent with other losses?

File: torch/nn/modules/loss.py [hunk 2]
```diff
@@ -458,9 +511,23 @@ class HingeEmbeddingLoss(_Loss):
...
 
-                         { x_i,                  if y_i ==  1
-        loss(x, y) = 1/n {
-                         { max(0, margin - x_i), if y_i == -1
+    The loss function for :math:`n`-th sample in the mini-batch is:
+
+    .. math::
+        l_n = \begin{cases}
+        x_n, & \text{if}\; y_n = 1,\\
+        \max \{0, \Delta - x_n\}, & \text{if}\; y_n = -1,
```
Comment (apaszke) [inline torch/nn/modules/loss.py]: can you please align the cases? this is useful when viewing the documentation inline in the shell

**Review by apaszke — COMMENTED** [review]
Looks good. Two minor things

**Review by soumith — APPROVED** [review]

**Comment by soumith** [comment]
thank you very much @zhouzq-thu !

**Pull request merged**